### ISSUE TITLE
clj-kondo :lint-as clojure.core/def causes false positive uninitialized var

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2510]
Repro:

``` clojure
(ns lint-as)

(defmacro my-def
  {:clj-kondo/lint-as 'clojure.core/def}
  [x]
  `(def ~x "dude"))

(my-def foo) uninitialized var

(def x "dude") ;; no warning
```

**Comment by borkdude** [comment]
This is expected since `(my-def foo)` is treated as `(def foo)` .
